presto-main/src/main/java/io/prestosql/spiller/FileSingleStreamSpillerFactory.java

```diff
@@ -74,27 +76,49 @@
     private final ListeningExecutorService executor;
     private final PagesSerdeFactory serdeFactory;
     private final List<Path> spillPaths;
+    private final boolean canSpill;
     private final SpillerStats spillerStats;
     private final double maxUsedSpaceThreshold;
     private final boolean spillEncryptionEnabled;
     private int roundRobinIndex;
     private final LoadingCache<Path, Boolean> spillPathHealthCache;
 
     @Inject
-    public FileSingleStreamSpillerFactory(Metadata metadata, SpillerStats spillerStats, FeaturesConfig featuresConfig, NodeSpillConfig nodeSpillConfig)
+    public FileSingleStreamSpillerFactory(
+            Metadata metadata,
+            SpillerStats spillerStats,
+            ServerConfig serverConfig,
+            NodeSchedulerConfig nodeSchedulerConfig,
+            FeaturesConfig featuresConfig,
+            NodeSpillConfig nodeSpillConfig)
     {
         this(
                 listeningDecorator(newFixedThreadPool(
                         requireNonNull(featuresConfig, "featuresConfig is null").getSpillerThreads(),
                         daemonThreadsNamed("binary-spiller-%s"))),
                 requireNonNull(metadata, "metadata is null").getBlockEncodingSerde(),
                 spillerStats,
-                requireNonNull(featuresConfig, "featuresConfig is null").getSpillerSpillPaths(),
+                getSpillPaths(
+                        requireNonNull(serverConfig, "serverConfig is null"),
+                        requireNonNull(nodeSchedulerConfig, "nodeSchedulerConfig is null"),
+                        requireNonNull(featuresConfig, "featuresConfig is null")),
                 requireNonNull(featuresConfig, "featuresConfig is null").getSpillMaxUsedSpaceThreshold(),
                 requireNonNull(nodeSpillConfig, "nodeSpillConfig is null").isSpillCompressionEnabled(),
                 requireNonNull(nodeSpillConfig, "nodeSpillConfig is null").isSpillEncryptionEnabled());
     }
 
+    private static List<Path> getSpillPaths(
+            ServerConfig serverConfig,
+            NodeSchedulerConfig nodeSchedulerConfig,
+            FeaturesConfig featuresConfig)
+    {
+        if (serverConfig.isCoordinator() && !nodeSchedulerConfig.isIncludeCoordinator()) {
+            return ImmutableList.of();
```

Comment:
This looks like you shouldn't configure spill paths on coordinator at all...

so what about https://github.com/prestosql/presto/blob/047d75056b856a9d5090bb145c1bf338d2173ec9/presto-main/src/main/java/io/prestosql/SystemSessionProperties.java#L368?

Reply:
I've also updated `SystemSessionProperties.java` code.

> This looks like you shouldn't configure spill paths on coordinator at all...

Coordinator might participate in running queries.

There is still a check `io.prestosql.sql.analyzer.FeaturesConfig#isSpillerSpillPathsConfiguredIfSpillEnabled` but it's harder to fix as we don't have all configuration to automatically validate it by aitlift. It would need to be at module initialization.

Reply:
So what about

if you `include-coordinator=false` then you just should not configure spiller-paths.

Then you don't need the if and  this method at all.
Maybe actually all you need is to remove the check in SSP?